### PR TITLE
1.13.1 rc1 cherrypicks

### PR DESCRIPTION
**Pull request by vieux**
Add zsh completion for 'docker {container,image,network,system} prune --filter #30317 
Add zsh completion for 'docker swarm --availability' #30282
Add bash completion for `dockerd --init` and `docker run|create --init` #30266
Add bash completion for `build --squash` #30200
Honour $DOCKER_HIDE_LEGACY_COMMANDS in bash completion #30132
Fix treatment of DOCKER_HIDE_LEGACY_COMMANDS in bash completion #30163
Add bash completion for `docker swarm unlock|unlock-key` #30399
Add missing options to bash completion for `docker swarm init|update` #30398
Add bash completion for `docker service ps --filter node` #30375
Fix bash completion for `docker service update` #30373
Add zsh completion for 'docker plugin install --alias' #30268
Add bash completion for `plugin install --disable-content-trust` #30125
Add zsh completion for 'dockerd --init' and 'docker {create,run} --init' #30318
Ensure proper value is used when computing reclaimable space #30378
Add bash completion and docs for `docker run|create --init-path` #30283
distribution: Add text/html and application/json as image mediatypes #30262
Remove timeout on fifos opening #30140
Ignore certificate expiry error for top-level inspect #29246
Fix pluginv1 Windows volumes #30150
Only show global warnings once #28939
do not create init-dir if not needed #29846
Fix deadlock on v1 plugin with activate error #30408
Fix failure in `docker ps --format` when `.Label` has args #30291
Fix bash completion for completing nodes #30376
plugins: add example docs for a rexray plugin #30294
[project] bump go to go 1.7.4 #29142
[ppc64le] use official go 1.7.4 binaries #29256
Windows: Bump to latest version of git #29159
Windows: Add service dependency ConDrv #30111
Fix incorrect `Scope` in `network ls/inspect` with duplicate network names #30250
Add "src" alias for `--secret` #30407
Fixed secret creation usage during service create in docs #30370
Remove attachable network on swarm leave #30157

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @albers @sdurrheimer if there's changes missing 😄

**Comment by sdurrheimer** [comment]
#30268

And I think I have one or two more things to add.

**Comment by albers** [comment]
#30375 was just merged.
#30125 is about a feature in 1.13.0, so it should go into 1.13.1. (I see missing support for a feature in completion as a bug.)

There are some pending PRs, so stay tuned: #30399, #30398, #30376, #30373, #30371, #30283. 
@thaJeztah maybe you can have a look at these?

**Comment by vieux** [comment]
@albers #30283 is not for 1.13.1

I added all the other that are merged

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -541,6 +561,7 @@ __docker_container_subcommand() {
         "($help)*--group=[Set one or more supplementary user groups for the container]:group:_groups"
         "($help -h --hostname)"{-h=,--hostname=}"[Container host name]:hostname:_hosts"
         "($help -i --interactive)"{-i,--interactive}"[Keep stdin open even if not attached]"
+        "($help)--init[Run an init inside the container that forwards signals and reaps processes]" \
```
Comment (thaJeztah) [inline contrib/completion/zsh/_docker]: This trailing backslash should be removed; I think there's a fix in another PR, but that also has other changes, so we should make sure that either that one, or a fix for just this trailing backslash makes it into 1.13.1

**Review by thaJeztah — DISMISSED** [review]

File: contrib/completion/zsh/_docker (hunk 1 above)
Comment (vieux) [inline contrib/completion/zsh/_docker]: @thaJeztah fixed in 328775ea4f8446394d45a333652d011f5006a360

**Comment by unclejack** [comment]
https://github.com/docker/docker/pull/30489 will also need to be cherry picked after the merge.

**Comment by tiborvass** [comment]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by vieux** [comment]
@unclejack yes I'll take care of it after

**Comment by vieux** [comment]
unrelated windows test error.

**Pull request merged**